tutorials/PreprocessingResnet50.ipynb

```diff
@@ -0,0 +1,675 @@
+    "    for i in range(nsamples):\n",
+    "        print(images[i].size)\n",
+    "    fig, axs = plt.subplots(1, nsamples)\n",
+    "    for i in range(nsamples):\n",
+    "        axs[i].axis('off')\n",
+    "        axs[i].imshow(images[i])\n",
+    "    plt.show()\n",
+    "\n",
+    "images = [\n",
+    "    Image.open('assets/dog.jpg'),\n",
+    "    Image.open('assets/cat.jpg')\n",
+    "]\n",
+    "show_images(images)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "96417141",
+   "metadata": {},
+   "source": [
+    "## Using an existing model\n",
+    "\n",
+    "For this example, we will use an existing ResNet50 model from the ONNX model repository "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "fc121ea2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import requests\n",
+    "import re\n",
+    "import json\n",
+    "\n",
+    "def get_sha_and_size(url):\n",
+    "    r = requests.get(url)\n",
+    "    assert r\n",
+    "    r_str = r.content.decode('utf-8')\n",
+    "    sha = re.findall('oid sha256:([A-Fa-f0-9]{64})', r_str)[0]\n",
+    "    size = re.findall('size (\\d+)', r_str)[0]\n",
+    "    return sha, size\n",
+    "\n",
+    "def get_href(sha, size, git_url='https://github.com/onnx/models.git'):\n",
+    "    headers = {\n",
+    "        'Accept': 'application/vnd.git-lfs+json',\n",
+    "        'Content-Type': 'application/json',\n",
+    "    }\n",
+    "    d = {'operation' : 'download', 'transfer' : ['basic'], 'objects' : [ {'oid': sha, 'size': int(size)}]}\n",
+    "    req = json.dumps(d)\n",
+    "    r = requests.post(f'{git_url}/info/lfs/objects/batch', headers=headers, data=req)\n",
+    "    assert r\n",
+    "    r_str = r.content.decode('utf-8')\n",
+    "    r_json = json.loads(r_str)\n",
+    "\n",
+    "    for obj in r_json['objects']:\n",
+    "        if obj['oid'] == sha and obj['size'] == int(size):\n",
+    "            href = obj['actions']['download']['href']\n",
+    "            return href\n",
+    "    return None\n",
+    "\n",
+    "def download_href(href, out_file):\n",
+    "    r = requests.get(href)\n",
+    "    assert r\n",
+    "    file_content = r.content\n",
+    "    with open(out_file, 'wb') as f:\n",
+    "        f.write(file_content)\n",
+    "\n",
+    "def download_lfs_file(url, out_file):\n",
+    "    org, repo = re.findall('https://raw.githubusercontent.com/(.*?)/(.*?)/', url)[0]\n",
+    "    sha, size = get_sha_and_size(url)\n",
+    "    href = get_href(sha, size, git_url=f'https://github.com/{org}/{repo}.git')\n",
+    "    download_href(href, out_file)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "84ae6f04",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "url = 'https://raw.githubusercontent.com/onnx/models/main/vision/classification/resnet/model/resnet50-v1-12.onnx'\n",
+    "out_file = 'resnet50-v1-12.onnx'\n",
+    "download_lfs_file(url, out_file)"
```

Comment:
nit: we can directly use onnx.hub to get the model
```
from onnx import hub
rn50_network = hub.load("ResNet50-fp32")
```

Reply:
That's great. Done